addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -78,9 +78,19 @@ options.registry.TableOfContent = options.Class.extend({
      * @private
      */
     _generateNav: function (ev) {
+        const targetedEls = Array.from(this.$target[0].querySelectorAll(this.targetedElements));
+        const blockTextContent = this.$target[0].textContent.replaceAll('\n', '').trim();
+        if (blockTextContent === '') {
+            // destroy public widget and remove the ToC since there are no more
+            // child elements, before doing so the observer needs to be
+            // disconnected else observer observe mutation and _generateNav
+            // gets called even after there's no more ToC.
+            this.observer.disconnect();
+            this.trigger_up('remove_snippet', {$snippet: this.$target});
+            return;
+        }
         this.options.wysiwyg && this.options.wysiwyg.odooEditor.unbreakableStepUnactive();
-        const headingsEls = this.$target.find(this.targetedElements).toArray()
-            .filter(el => !el.closest('.o_snippet_desktop_invisible'));
+        const headingsEls = targetedEls.filter(el => !el.closest('.o_snippet_desktop_invisible'));
```

Comment:
As we are in stable version, we try to do a minimal change. You don't need this change anymore and you don't need the new variable `targetedEls` right ? This kind of change is welcome in master but not in stable. Can you remove those unnecessary changes please.